flash/core/data/utilities/paths.py

```diff
@@ -152,18 +153,37 @@ def filter_valid_files(
     if not isinstance(files, List):
         files = [files]
 
+    if valid_extensions is None:
+        return (files,) + additional_lists
+
+    if not isinstance(valid_extensions, tuple):
+        valid_extensions = tuple(valid_extensions)
+
     additional_lists = tuple([a] if not isinstance(a, List) else a for a in additional_lists)
 
     if not all(len(a) == len(files) for a in additional_lists):
         raise MisconfigurationException(
             f"The number of files ({len(files)}) and the number of items in any additional lists must be the same."
         )
 
-    if valid_extensions is None:
-        return (files,) + additional_lists
     filtered = list(
         filter(lambda sample: has_file_allowed_extension(sample[0], valid_extensions), zip(files, *additional_lists))
     )
+
     if len(additional_lists) > 0:
         return tuple(zip(*filtered))
-    return [f[0] for f in filtered]
+
+    filtered = [f[0] for f in filtered]
+
+    invalid = [f for f in files if f not in filtered]
+
+    if invalid:
+        invalid_extensions = list({"." + f.split(".")[-1] for f in invalid})
+        invalid_extensions_message = ", ".join(invalid_extensions)
+        valid_extensions_message = ", ".join(valid_extensions)
+        rank_zero_warn(
+            f"Found invalid file extensions: {invalid_extensions_message}"
+            f"The supported file extensions are: {valid_extensions_message}"
+        )
```

Comment:
Small simplification:
```suggestion
        rank_zero_warn(
            f"Found invalid file extensions: {', '.join(invalid_extensions)}"
            f"The supported file extensions are: {', '.join(valid_extensions)}"
        )
```

Reply:
resolved in https://github.com/Lightning-AI/lightning-flash/pull/1386/commits/809b93c6d85547b7196c1a7420b41a5fd4af6bad